lib/spack/llnl/util/cpu.py

```diff
@@ -0,0 +1,316 @@
+        parents = [targets.get(p) for p in parents]
+
+        # Get target vendor
+        vendor = values.get('vendor', None)
+        if not vendor:
+            vendor = parents[0].vendor
+
+        features = set(values['features'])
+        compilers = values.get('compilers', {})
+        generation = values.get('generation', 0)
+
+        targets[name] = MicroArchitecture(
+            name, parents, vendor, features, compilers, generation
+        )
+
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    filename = os.path.join(this_dir, 'targets.json')
+    with open(filename, 'r') as f:
+        data = json.load(f)
+
+    targets = OrderedDict()
+    for name in data:
+        if name in targets:
+            # name was already brought in as ancestor to a target
+            continue
+        fill_target_from_dict(name, data, targets)
+
+    # Add the host platform if not present
+    host_platform = platform.machine()
+    targets.setdefault(host_platform, create_generic_march(host_platform))
+
+    return targets
+
+
+#: Dictionary of known micro-architectures
+targets = _load_targets_from_json()
+
+
+def supported_target_names():
+    return targets.keys()
+
+
+def _create_cpuinfo_dict():
+    """Returns a dictionary with information on the host CPU."""
+    dict_factory = {
+        'Linux': _create_dict_from_proc,
+        'Darwin': _create_dict_from_sysctl
+    }
+    return dict_factory[platform.system()]()
+
+
+def _create_dict_from_proc():
+    # Initialize cpuinfo from file
+    cpuinfo = {}
+    try:
+        with open('/proc/cpuinfo') as file:
+            text = file.readlines()
+            for line in text:
+                if line.strip():
+                    key, _, value = line.partition(':')
+                    cpuinfo[key.strip()] = value.strip()
+    except IOError:
+        return None
+    return cpuinfo
+
+
+def _create_dict_from_sysctl():
+
+    def check_output(args):
+        if sys.version_info >= (3, 0):
+            return subprocess.run(
+                args, check=True, stdout=subprocess.PIPE).stdout  # nopyqver
+        else:
+            return subprocess.check_output(args)  # nopyqver
+
+    cpuinfo = {}
+    try:
+        cpuinfo['vendor_id'] = check_output(
+            ['sysctl', '-n', 'machdep.cpu.vendor']
+        ).strip()
+        cpuinfo['flags'] = check_output(
+            ['sysctl', '-n', 'machdep.cpu.features']
+        ).strip().lower()
+        cpuinfo['flags'] += ' ' + check_output(
+            ['sysctl', '-n', 'machdep.cpu.leaf7_features']
+        ).strip().lower()
+        cpuinfo['model'] = check_output(
+            ['sysctl', '-n', 'machdep.cpu.model']
+        ).strip()
+        cpuinfo['model name'] = check_output(
+            ['sysctl', '-n', 'machdep.cpu.brand_string']
+        ).strip()
+
+        # Super hacky way to deal with slight representation differences
+        # Would be better to somehow consider these "identical"
```

Comment:
I left this code as it was. I did that because with the refactor in 7a6e87b, due to the decorator that is now being used, it's imo clear for a reader that this hack is peculiar to `Darwin`. 

Also note that here we are collecting raw information in a dictionary to be used later when detecting the micro-architecture for the host. This part of the package should be considered either advanced or part of the internal logic and as such it's not exposed to normal users.

To avoid users having to remember that SSE 4.1 it's spelled `sse4_1` when performing operations like:
```python
`sse4_1` in target
```
I added new aliases in the JSON file, see 981a25e. Let me know if this is fine or needs further refinement.

Reply:
@tgamblin Are you fine with :point_up: ?

Reply:
I like the alias-based solution!